relay-statsd/src/lib.rs

```diff
@@ -85,6 +96,27 @@ pub struct MetricsClient {
     pub rx: Option<crossbeam_channel::Receiver<Vec<u8>>>,
 }
 
+/// Client configuration used for initialization of [`MetricsClient`].
+#[derive(Debug)]
+pub struct MetricsClientConfig<'a, A: ToSocketAddrs> {
+    /// Prefix which is appended to all metric names.
+    pub prefix: &'a str,
+    /// Host of the metrics upstream.
+    pub host: A,
+    /// Tags that are added to all metrics.
+    pub default_tags: BTreeMap<String, String>,
+    /// Sample rate for metrics, between 0.0 (= 0%) and 1.0 (= 100%)
+    pub sample_rate: f32,
+    /// If metrics should be batched or send immediately upstream.
+    pub aggregate: bool,
+    /// A list of tag names that will be removed when they exactly match any value specified.
+    pub deny_tags: Vec<String>,
+    /// A list of prefixes. Any tag that ends with any of these suffixes will be removed.
+    pub deny_starts_with: Vec<String>,
+    /// A list of suffixes. Any tag that ends with any of these suffixes will be removed.
+    pub deny_ends_with: Vec<String>,
```

Comment:
You could use either your own `DenyMetricTags` struct or `statsdproxy`'s `DenyTagConfig` struct here, but it's fine either way.